src/Parlot/Scanner.cs

```diff
@@ -472,5 +472,56 @@ private bool ReadQuotedString(char quoteChar, out TokenResult result)
 
             return true;
         }
+
+        /// <summary>
+        /// Reads a sequence token enclosed in arbritrary start and end characters.
+        /// </summary>
+        /// <remarks>
+        /// This method doesn't escape the string, but only validates its content is syntactically correct.
+        /// The resulting Span contains the original quotes.
+        /// </remarks>
+        public bool ReadNonEscapableSequence(char startSequenceChar, char endSequenceChar, out TokenResult result)
+        {
+            var startChar = Cursor.Current;
+
+            if (startChar != startSequenceChar)
+            {
+                result = TokenResult.Fail();
+                return false;
+            }
+
+            // Fast path if there aren't any escape char until next quote
+            var startOffset = Cursor.Offset + 1;
+            var lastQuote = startOffset;
+
+            int nextQuote ;
+            do
+            {
+                nextQuote = Cursor.Buffer.IndexOf(endSequenceChar, lastQuote + 1);
+
+                if (nextQuote == -1)
+                {
+                    if(startOffset == lastQuote)
+                    {
+                        // There is no end sequence character, not a valid escapable sequence
+                        result = TokenResult.Fail();
+                        return false;
+                    }
+                    nextQuote = lastQuote - 1;
+                    break;
+                }
+
+                lastQuote = nextQuote + 1;
+            }
+            while(Cursor.Buffer.Length > lastQuote && Cursor.Buffer[lastQuote] == endSequenceChar);
+
+            var start = Cursor.Position;
+
+            // If the next escape if not before the next quote, we can return the string as-is
+            Cursor.Advance(nextQuote + 2 - startOffset);
```

Comment:
It's a nice optimization to call Advance only once, but then you are missing on `EOF` checks. So a string like `'foo` will fail.  Which shows some unit tests are missing.